src/bridge/header.ts

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import { useEffect, useMemo } from 'react';
+
+interface BreadcrumbsPayload {
+  type: 'header.breadcrumbs';
+  items: {
+    label: string;
+    onClick: string;
+  }[];
+}
+
+interface PrimaryCtaPayload {
+  type: 'header.primaryCta';
+  label: string;
+  onClick: string;
+}
+
+interface Breadcrumb {
+  label: string;
+  onClick?: () => void;
+}
+
+const getBreadcrumbId = (idx: number) => `header.breadcrumbs.${idx}`;
+
+export function useBreadcrumbs(breadcrumbs: Breadcrumb[]) {
+  const callbackRefs = useMemo(() => {
+    return breadcrumbs.reduce<Record<string, () => void>>(
+      (acc, { onClick }, idx) => {
+        if (onClick) acc[getBreadcrumbId(idx)] = onClick;
+        return acc;
+      },
+      {},
+    );
+  }, [breadcrumbs]);
+
+  const payload: BreadcrumbsPayload = {
+    type: 'header.breadcrumbs',
+    items: breadcrumbs.map(({ label, onClick }, idx) => ({
+      onClick: onClick ? getBreadcrumbId(idx) : '',
+      label,
+    })),
+  };
+
+  useEffect(() => {
+    window.parent.postMessage(payload, 'https://dashboard.copilot.com');
+
+    // Be sure to add your portal domain here as well, whether it's a copilot.app
+    // subdomain or a custom domain. This allows the two frames to talk to each other.
+    // window.parent.postMessage(payload, 'https://yourportaldomain.copilot.app');
```

Comment:
this need to be dynamically set right? Because the domain can be different based on which workspace the app is installed on.

Reply:
Good call, I added that logic.